### ISSUE TITLE
Switch from S3 to Vulr object storage

### ISSUE THREAD

**Issue by starbelly** [erlef/website#169]
https://www.vultr.com/docs/vultr-object-storage

**Comment by joaquinalcerro** [comment]
@starbelly,

I am going to try tackle this issue but have some questions:

1. Do you want to completely replace the AWS S3 Storage with the Vultr Storage or do you want to have the possibility to use both?
2. What MIME file types should be allowed?
3. What is the max size? I understand a 8MB is max allowed by Plug.

Thanks

**Comment by joaquinalcerro** [comment]
@starbelly 

So this is what I have for now:
1. Looks like Vultr Object Storage is AWS S3 "compatible" which means you can call Vultr's API with same AWS S3 libraries or applications with minimum configuration.
2. I configured a personal account in Vultr and created a bucket "events-org-images"
3. I made minimum configuration changes to the application:

```
config :ex_aws,
  s3: [
    # region: "us-east-2",
    scheme: "https://",
    region: "New Jersey",
    host: "ewr1.vultrobjects.com"
  ]

config :ex_aws,
  access_key_id: [System.get_env("VULTR_ACCESS_KEY_ID"), :instance_role],
  secret_access_key: [System.get_env("VULTR_SECRET_KEY_ID"), :instance_role]
```
4. Change the Storage Module to use the new URL to access the uploaded file
```
    case ExAws.request(operation) do
      {:ok, _} ->
        # {:ok, "https://event-org-images.s3.us-east-2.amazonaws.com/#{filename}"}
        {:ok, "https://event-org-images.ewr1.vultrobjects.com/#{filename}"}

      err ->
        err
    end
```
5. When I call the function Erlef.Storage.upload_event_org_image/3

```
iex(8)> alias Erlef.Storage
Erlef.Storage
iex(9)> path = "solve_problems.txt"
"solve_problems.txt"
iex(10)> binary = File.read!(path)
"Problemas que Cloud Computing resuelve:\n1. Almacenamiento de informacion:\n    a. Fisicamente donde colocar la informacion - legalmente\n    b. Grandes volumenes de informacion requiere mucho especion local\n    c. Procesar grandes volumnes de .................
iex(11)> Storage.upload_event_org_image(path, binary)
{:ok, "https://event-org-images.ewr1.vultrobjects.com/solve_problems.txt"}
iex(12)>
```

So, look like it is working for this functionality but will like your feedback to proceed with it.

Still pending:
1. Your feedback of my first post.
2. Your feedback about this approach

Best regards,

**Comment by starbelly** [comment]
Hey @joaquinalcerro Yeah, things have been interesting :) I've made a reminder to circle back to this by EOD.

**Comment by starbelly** [comment]
@joaquinalcerro Yeah, looks solid. I'd say do a PR 👍

**Comment by starbelly** [comment]
Closed via #192
